controllers/opentelemetrycollector_controller.go

```diff
@@ -66,7 +66,7 @@ func NewReconciler(p Params) *OpenTelemetryCollectorReconciler {
 		p.Tasks = []Task{
 			{
 				"config maps",
-				reconcile.ConfigMaps,
+				reconcile.CollectorConfigMaps,
```

Comment:
The reconcile package shouldn't be aware of how to reconcile configmaps for collector or for TA, only "config maps" in general. The list of configmaps to reconcile would be provided by the collector or TA logic, but the reconciliation itself (desired vs. existing logic) is handled by the common code. If you need a special logic in one of them, then you implement it in a specialized reconciler.

Reply:
Do you think it would be better to have this refactoring as part of another PR?

Reply:
Resolving this since it is being handled in another PR and the new updates based on that will follow.